File: tooltaoanh/flow_login.py
from playwright.sync_api import sync_playwright
import time
import os

def main():
    profile_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'flow_profile')
    print("Dữ liệu đăng nhập Google Flow sẽ được lưu tại:", profile_dir)
    
    with sync_playwright() as p:
        browser = p.chromium.launch_persistent_context(
            user_data_dir=profile_dir,
            headless=False,
            viewport={'width': 1280, 'height': 800},
            args=['--disable-blink-features=AutomationControlled']
        )
        
        page = browser.pages[0] if browser.pages else browser.new_page()
        page.goto('https://labs.google/fx/vi/tools/flow')
        
        print("\n" + "="*50)
        print("✅ ĐÃ MỞ TRÌNH DUYỆT THÀNH CÔNG!")
        print("-> Anh hãy đăng nhập Google trên cửa sổ trình duyệt vừa hiện lên.")
        print("-> Thông tin đăng nhập sẽ được lưu lại cho các lần chạy tool tự động sau này.")
        print("-> Cửa sổ này sẽ giữ mở trong 1 giờ để anh thong thả thao tác.")
        print("-> Khi đã đăng nhập thành công và thấy giao diện Flow, hãy báo em trên này nhé!")
        print("="*50 + "\n")
        
        try:
            # Hold browser open for 1 hour
            time.sleep(3600)
        except Exception:
            pass
        finally:
            browser.close()

if __name__ == "__main__":
    main()

File: LandscapeApp/server/flowLogin.js
const path = require('path');
const { chromium } = require('playwright-core');

const FLOW_PROFILE_DIR = path.resolve(__dirname, '..', '..', 'tooltaoanh', 'flow_profile');

const CHROME_CANDIDATES = [
  'C:\\Program Files\\Google\\Chrome\\Application\\chrome.exe',
  'C:\\Program Files (x86)\\Google\\Chrome\\Application\\chrome.exe',
  'C:\\Program Files\\Microsoft\\Edge\\Application\\msedge.exe',
  'C:\\Program Files (x86)\\Microsoft\\Edge\\Application\\msedge.exe'
];

function resolveBrowserExecutable() {
  const executablePath = CHROME_CANDIDATES.find(candidate => require('fs').existsSync(candidate));
  if (!executablePath) {
    throw new Error('Không tìm thấy Chrome hoặc Edge trên máy.');
  }
  return executablePath;
}

async function main() {
  console.log("==================================================");
  console.log("   KHỞI ĐỘNG CỔNG ĐĂNG NHẬP GOOGLE LABS FLOW...");
  console.log("==================================================");
  console.log("-> Nơi lưu phiên đăng nhập: ", FLOW_PROFILE_DIR);
  
  let browser;
  try {
    browser = await chromium.launchPersistentContext(FLOW_PROFILE_DIR, {
      executablePath: resolveBrowserExecutable(),
      headless: false,
      viewport: { width: 1440, height: 900 },
      args: ['--disable-blink-features=AutomationControlled']
    });

    const page = browser.pages().length > 0 ? browser.pages()[0] : await browser.newPage();
    
    console.log("-> Đang mở trang Google Labs Flow...");
    await page.goto('https://labs.google/fx/vi/tools/flow');

    console.log("\n✅ ĐÃ MỞ TRÌNH DUYỆT THÀNH CÔNG!");
    console.log("-> Anh hãy tiến hành đăng nhập bằng tài khoản Google.");
    console.log("-> Nhấn 'Tiếp tục' / Chấp nhận điều khoản nếu có.");
    console.log("-> Cửa sổ này sẽ không tự tắt trong vòng 1 tiếng để anh thoải mái thao tác.");
    console.log("-> Sau khi anh thấy VÀO ĐƯỢC GIAO DIỆN FLOW thành công, HÃY NHẮN BÁO CHO EM NHÉ!");
    
    // Giữ mở 1 giờ (3600000 ms)
    await new Promise(r => setTimeout(r, 3600000));
    
  } catch (err) {
    console.error("LỖI:", err);
  } finally {
    if (browser) {
      await browser.close().catch(() => {});
    }
  }
}

main();
